src/persist/src/indexed/mod.rs

```diff
@@ -197,6 +245,13 @@ impl<U: Buffer, L: Blob> Indexed<U, L> {
         self.traces
             .entry(id)
             .or_insert_with_key(|id| BlobTrace::new(BlobTraceMeta::new(*id)));
+
+        if let Err(e) = self.blob.set_meta(self.serialize_meta()) {
+            // We were unable to durably commit the register. Revert back to the
+            // previous version of meta.
+            self.restore(prev);
```

Comment:
worth pulling out a method for this whole `if let Err(e) = self.blob.set_meta(self.serialize_meta()) { ...; self.restore(prev); ...}` bit? we could also put the compactable log TODO in there and keep it from being copied everywhere